pkg/config/draftconfig.go

```diff
@@ -258,6 +316,36 @@ func (d *DraftConfig) ApplyDefaultVariablesForVersion(version string) error {
 	return nil
 }
 
+func (d *DraftConfig) CheckActiveWhenConstraint(refVar *BuilderVar, activeWhen ActiveWhenConstraint) (bool, error) {
+	checkValue := refVar.Value
+	if checkValue == "" {
+		if refVar.Default.Value != "" {
+			checkValue = refVar.Default.Value
+		}
+
+		if refVar.Default.ReferenceVar != "" {
+			refValue, err := d.recurseReferenceVars(refVar, refVar, true)
+			if err != nil {
+				return false, err
+			}
+			if refValue == "" {
+				return false, errors.New("reference variable has no value")
+			}
+
+			checkValue = refValue
+		}
+	}
+
+	switch activeWhen.Condition {
+	case EqualTo:
+		return checkValue == activeWhen.Value, nil
+	case NotEqualTo:
+		return checkValue != activeWhen.Value, nil
+	}
+
+	return false, nil
```

Comment:
should we return an error for unrecognized condition here?

Reply:
added